support-frontend/assets/pages/[countryGroupId]/components/retryPaymentStatus.ts

```diff
@@ -0,0 +1,52 @@
+import type { StatusResponse } from 'helpers/forms/paymentIntegrations/readerRevenueApis';
+
+// ----- Setup ----- //
+const DEFAULT_POLLING_INTERVAL_MILLIS = 3000;
+const DEFAULT_MAX_POLLS = 10;
+
+// corrects ES5 instanceOf error for retry mechanism
+// https://dev.to/dguo/how-to-fix-instanceof-not-working-for-custom-errors-in-typescript-4amp
+class StillPendingError extends Error {
+	constructor(message: string, public trackingUri: string) {
+		super(message);
+		Object.setPrototypeOf(this, StillPendingError.prototype);
+		this.trackingUri = trackingUri;
+	}
+}
+
+function timeOut(milliseconds: number | undefined): Promise<void> {
+	return new Promise((resolve) => setTimeout(resolve, milliseconds));
+}
+
+export function retryPaymentStatus(
+	promiseFunction: () => Promise<StatusResponse>,
+	pollMax: number = DEFAULT_MAX_POLLS,
+	pollInterval: number = DEFAULT_POLLING_INTERVAL_MILLIS,
+	onRetry?: (pollCount: number, pollTimeTotal: number) => void,
```

Comment:
I can't see anywhere where this parameter is ever passed in, do we need it?

Reply:
Good point, was used for testing purposes (notify of a `pending` retry), removed now.